contracts/axelar-gateway/src/testutils.rs

```diff
@@ -60,12 +60,23 @@ pub fn get_approve_hash(env: &Env, messages: Vec<Message>) -> BytesN<32> {
 }
 
 pub fn generate_test_message(env: &Env) -> (Message, Bytes) {
-    let mut rng = rand::thread_rng();
-    let len = rng.gen_range(0..20);
-    let mut payload = std::vec![0u8; len];
-    rng.fill(&mut payload[..]);
+    generate_test_message_with_randomness(env, true)
+}
 
-    let payload = Bytes::from_slice(env, &payload[..]);
+pub fn generate_deterministic_test_message(env: &Env) -> (Message, Bytes) {
+    generate_test_message_with_randomness(env, false)
+}
+
+fn generate_test_message_with_randomness(env: &Env, use_rng: bool) -> (Message, Bytes) {
```

Comment:
This is a weird function signature, the name indicates randomness is used, but the arg allows configuring it. Perhaps create two separate functions:

`dummy_message` and `generate_message` with separate implementations since duplication is very minor and the values don't technically need to match between the two anyways